### PR TITLE
Removes the ability to attack with weapons while prone

### PR DESCRIPTION
**Pull request by Wisemonster**
## About The Pull Request
Title, removes the ability to attack with weapons if you are prone. This should include things like getting stunned from a pounce, or simply lying down.
## Why It's Good For The Game
Melee weapons being able to attack while prone has been a long standing bug, and I'm gonna finally fix it.
## Changelog
:cl:
fix: Can no longer attack with melee weapons while lying down.
/:cl:

### CODE REVIEW

**Comment by Wisemonster** [comment]
To clarify, this should only really affect weapons with straps attached to them, such as vali weapons and the powerfist.

**Review by Lumipharon — COMMENTED** [review]
Is this actually specific to weapon items?

**Comment by Wisemonster** [comment]
The shields are considered weapons, and I don't believe there are any non-weapon items that can be held while lying down (aka. have straps)

**Comment by Lumipharon** [comment]
Straps are not inherent to weapon items. Just because you can't strap a welder or something today doesn't mean it won't be the case in the future.

**Comment by Wisemonster** [comment]
Alright, I moved it to the parent proc. If you'd rather me move it to the strap component, I can try to do that.

File: code/_onclick/item_attack.dm [hunk 1]
```diff
@@ -5,6 +5,9 @@
 		return melee_attack_chain_alternate(user, target, params)
 	if(tool_behaviour && tool_attack_chain(user, target))
 		return
+	if(user.lying_angle == 90 || user.lying_angle == 270)
+		user.balloon_alert(user, "Can't while prone!")
+		return
```
Comment (TiviPlus) [inline code/_onclick/item_attack.dm:10]:
```suggestion
	if(user.lying_angle)
		user.balloon_alert(user, "Can't while prone!")
		return
```

**Review by TiviPlus — COMMENTED** [review]
more encompassing fix, we use this often to check the same thing

**Review by Lumipharon — APPROVED** [review]

**Comment by Looking4bros** [comment]
it's so joever swordbros

**Comment by Looking4bros** [comment]
Is this really a bug? After having my legs cut off several times i feel that this is really just a nerf. The only weapon that can be used one handed are the one handed swords, which are already not used when two handed options are available. Or in HvH, but that's HvH and speed is more needed than added power.

This has rendered me completely helpless to the whims of the xenos i'm fighting. Before this, i could at least hit them weakly with the claymore, and encourage them to kill me, rather than drag me wherever they want to.

**Pull request merged**